api/client_test.go

```diff
@@ -152,7 +152,7 @@ func BenchmarkClient(b *testing.B) {
 			}
 			b.ResetTimer()
 			for i := 0; i < b.N; i++ {
-				_, _, err := client.Do(ctx, req)
+				_, err := client.Do(ctx, req)
```

Comment:
This is now starting the call but never reading the response. It's no longer benchmarking anything interesting.

`BenchmarkClient` was added to illustrate the difference between ways of filling the buffer; To illustrate the difference between buffer and no buffer you have to go to to a higher level and benchmark reading an entire Prometheus API response, including JSON parsing.